fuel-tx/src/transaction/types/create.rs

```diff
@@ -172,6 +173,32 @@ impl Chargeable for Create {
         // is defined. Witness data should still be excluded.
         self.witnesses_offset()
     }
+
+    fn gas_used_by_metadata(&self, gas_costs: &GasCosts) -> Word {
+        let Create {
+            bytecode_witness_index,
+            witnesses,
+            storage_slots,
+            ..
+        } = self;
+
+        let contract: Option<Contract> = witnesses
+            .get(*bytecode_witness_index as usize)
+            .map(|c| c.as_ref().into());
+
+        let contract_root_gas = contract
+            .map(|contract| {
+                let contract_len = contract.len() as Word;
+                gas_costs.contract_root.resolve(contract_len)
+            })
+            .unwrap_or(0);
+        let state_root_length = storage_slots.len() as Word;
+        let state_root_gas = gas_costs.state_root.resolve(state_root_length);
+
+        let contract_id_gas = gas_costs.contract_id;
```

Comment:
`contract_id` calculation is simply the calling of the hash functions. You can use dependent cost for hash function instead